adapter.go

```diff
@@ -82,21 +76,65 @@ func finalizer(a *Adapter) {
 }
 
 // NewAdapter is the constructor for Adapter.
+// Params : databaseName,tableName,dbSpecified
+//			databaseName,{tableName/dbSpecified}
+//			{database/dbSpecified}
+// databaseName and tableName are user defined.
+// Their default value are "casbin" and "casbin_rule"
+//
 // dbSpecified is an optional bool parameter. The default value is false.
 // It's up to whether you have specified an existing DB in dataSourceName.
 // If dbSpecified == true, you need to make sure the DB in dataSourceName exists.
-// If dbSpecified == false, the adapter will automatically create a DB named "casbin".
-func NewAdapter(driverName string, dataSourceName string, dbSpecified ...bool) (*Adapter, error) {
+// If dbSpecified == false, the adapter will automatically create a DB named databaseName.
+func NewAdapter(driverName string, dataSourceName string,params ...interface{}) (*Adapter, error) {
 	a := &Adapter{}
 	a.driverName = driverName
 	a.dataSourceName = dataSourceName
 
-	if len(dbSpecified) == 0 {
-		a.dbSpecified = false
-	} else if len(dbSpecified) == 1 {
-		a.dbSpecified = dbSpecified[0]
-	} else {
-		return nil, errors.New("invalid parameter: dbSpecified")
+	a.tableName = "casbin_rule"
+	a.databaseName = "casbin"
+	a.dbSpecified = false
+
+	if len(params) == 0 {
+
+	}else if len(params) == 1 {
+		switch p1:=params[0].(type) {
+		case bool:
+			a.dbSpecified = p1
```

Comment:
@rico-ci If I understand correctly, you can just pass in a bool variable as the value of dbSpecified.

Reply:
When there is no databseName or tableName on the old version, you can set dbspecified separately. So if you dont set dbname and tablename, it will use the default values.
Default values are set in adapter.go#L94-L96

Reply:
On the other hand, it is also compatible with older versions

Reply:
@rico-ci Do you have any question?